### PR TITLE
feat: [#507] Add `Abort` method for Response

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

RelatedTo https://github.com/goravel/goravel/issues/507

@coderabbitai summary

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

File: contracts/http/response.go [hunk 1]
```diff
@@ -49,6 +49,11 @@ type Response interface {
 	Render() error
 }
 
+type AbortResponse interface {
+	Response
+	Abort() error
```
Comment (hwbrzzl) [inline contracts/http/response.go:54]: Any concerns about not adding `Abort` to `Response` directly?

Comment (kkumar-gcc) [reply contracts/http/response.go:54]: I decided not to implement the abort method for streaming, downloads, and other formats as it was unnecessary. Currently, it is implemented only for `StringResponse` and `JsonResponse`, which makes more sense to me.

Comment (hwbrzzl) [reply contracts/http/response.go:54]: I think `Data` and `Redirect` are needed, too.

File: contracts/http/response.go [hunk 2]
```diff
@@ -49,6 +49,11 @@ type Response interface {
 	Render() error
 }
 
+type AbortResponse interface {
```
Comment (hwbrzzl) [inline contracts/http/response.go:52]:
```suggestion
type ResponseWithAbort interface {
```

**Review by hwbrzzl — DISMISSED** [review]
LGTM

**Review by hwbrzzl — APPROVED** [review]

**Pull request merged**